### PR TITLE
Product Editor Onboarding: Add "tell me more" button to wc.com page

### LINKED ISSUE #38518: Product Editor Onboarding: Add "tell me more" button to wc.com page
**Description**

Blocked by: https://github.com/woocommerce/woocommerce/issues/38516 and having a wc.com link.

We want to allow users to read more about the new product editor on an external documentation page, we will allow users to link to this on the 4th step of the feature tour by clicking "Tell me more".
This work will be dependant on https://github.com/woocommerce/woocommerce/issues/38516

**Design**

lj4AUMJnpMtUe7XqyvOONq-fi-1335_125284

**Acceptance criteria**

- [ ] "Tell me more" primary button should be displayed in tour on last (4th) step.
- [ ] Clicking "Tell me more" should link to wc.com page: 
https://woocommerce.com/product-form-beta
- [ ] Add tracking (see below for tracks)

**Tracking**

- User clicks tell me more - `product_editor_spotlight_tell_me_more_click`

### PR DESCRIPTION
**Pull request by nathanss**
### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Add "tell me more" button to wc.com page.
 
On top of that, I copied the Guide component from `@wordpress/components`, because I had to implement some things on it, as described in the comment left in the code.

I also slightly tweaked the place where the option `woocommerce_block_product_tour_shown` is set. It is only set when the guide is closed or finished.

Closes #38518 .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Enable the block product editor in WooCommerce > Settings > Advanced > Features.
2. Go to Product > Add New.
3. You should see the tour kit step in the bottom left of the editor. (If you have already dismissed it, you can delete the `woocommerce_block_product_tour_shown` option through WCA Test Helper)
4. Click 'View highlights'
5. Navigate through the guide
6. Check that the Close button is not focused every time the page switches
7. Close the Guide by clicking the Close button.
8. Verify that the `wcadmin_block_product_editor_spotlight_dismissed` tracks event was fired with the correct `current_page`.
9. Delete the `woocommerce_block_product_tour_shown` through the WCA Test Helper and refresh the page.
10. Go through the Guide again until the end and click 'Tell me more'
11. Verify that a new tab was opened with more information.

### CODE REVIEW

File: plugins/woocommerce-admin/client/products/tour/components/guide/index.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+							onClick={ goForward }
+						>
+							{ __( 'Next', 'woocommerce' ) }
+						</Button>
+					) }
+					{ ! canGoForward && (
+						<Button
+							className="components-guide__finish-button"
+							href={ finishButtonLink }
+							target={ finishButtonLink ? '_blank' : undefined }
```
Comment (mdperez86) [inline plugins/woocommerce-admin/client/products/tour/components/guide/index.tsx:110]: I would like to suggest [this](https://developer.chrome.com/en/docs/lighthouse/best-practices/external-anchors-use-rel-noopener/) when targeting blank.

**Review by mdperez86 — CHANGES_REQUESTED** [review]
I left one comment here. Everything is testing well @nathanss.

**Comment by nathanss** [comment]
@mdperez86 thanks for the suggestion. I implemented it!

**Review by mdperez86 — APPROVED** [review]
LGTM!!!

**Pull request merged**